src/Umbraco.Web/Templates/TemplateUtilities.cs

```diff
@@ -35,262 +29,28 @@ internal static string ParseInternalLinks(string text, bool preview, UmbracoCont
-                        text = text.Replace(url, resolvedUrl);
-                    }
-                }
-            }
-
-            return text;
-        }
+            => Current.Factory.GetInstance<UrlParser>().EnsureUrls(text);
 
+        [Obsolete("Use StringExtensions.CleanForXss instead")]
         public static string CleanForXss(string text, params char[] ignoreFromClean)
-        {
-            return text.CleanForXss(ignoreFromClean);
-        }
+            => text.CleanForXss(ignoreFromClean);
 
-        /// <summary>
-        /// Parses the string looking for Umbraco image tags and updates them to their up-to-date image sources.
-        /// </summary>
-        /// <param name="text"></param>
-        /// <returns></returns>
-        /// <remarks>Umbraco image tags are identified by their data-udi attributes</remarks>
+        [Obsolete("Use MediaParser.EnsureImageSources instead")]
         public static string ResolveMediaFromTextString(string text)
-        {
-            // don't attempt to proceed without a context
-            if (Current.UmbracoContext == null || Current.UmbracoContext.Media == null)
-            {
-                return text;
-            }
-
-            return ResolveImgPattern.Replace(text, match =>
-            {
-                // match groups:
-                // - 1 = from the beginning of the image tag until src attribute value begins
-                // - 2 = the src attribute value excluding the querystring (if present)
-                // - 3 = anything after group 2 and before the data-udi attribute value begins
-                // - 4 = the data-udi attribute value
-                // - 5 = anything after group 4 until the image tag is closed
-                var udi = match.Groups[4].Value;
-                if(udi.IsNullOrWhiteSpace() || GuidUdi.TryParse(udi, out var guidUdi) == false)
-                {
-                    return match.Value;
-                }
-                var media = Current.UmbracoContext.Media.GetById(guidUdi.Guid);
-                if(media == null)
-                {
-                    // image does not exist - we could choose to remove the image entirely here (return empty string),
-                    // but that would leave the editors completely in the dark as to why the image doesn't show
-                    return match.Value;
-                }
-
-                var url = media.Url;
-                return $"{match.Groups[1].Value}{url}{match.Groups[3].Value}{udi}{match.Groups[5].Value}";
-            });
-        }
-
-        /// <summary>
-        /// Removes media urls from &lt;img&gt; tags where a data-udi attribute is present
-        /// </summary>
-        /// <param name="text"></param>
-        /// <returns></returns>
+            => Current.Factory.GetInstance<ImageSourceParser>().EnsureImageSources(text);
+        
+        [Obsolete("Use MediaParser.RemoveImageSources instead")]
```

Comment:
Again this is renamed.. IIRC we can use `nameof(ImageSourceParser)` in locations like this.